sysl2/sysl/exporter/endpoint_exporter.go

```diff
@@ -130,41 +130,52 @@ func (e *EndpointExporter) setHTTPMethod(path string, endpoint *proto.Endpoint,
 	return op
 }
 
+func (e *EndpointExporter) setCommonAttributes(
+	name string,
+	param *spec.Parameter,
+	attrMap map[string]*proto.Attribute,
+	valueMap protoType,
+) {
+	param.Format = valueMap.Format
+	param.Type = valueMap.Type
+	param.Name = name
+	if _, ok := attrMap["required"]; ok {
+		param.Required = true
+	} else {
+		param = param.AsOptional()
+	}
+	if param.Type == object {
+		param.Schema.ExtraProps["$ref"] = "#/definitions/" + param.Format
+	}
+}
+
 func (e *EndpointExporter) setPathParams(endpoint *proto.Endpoint, op *spec.Operation) error {
+	var attrMap map[string]*proto.Attribute
```

Comment:
unnecessary, attrMap only used inside the loop

Reply:
for range loop in golang doesn't allow you to create and initialise new variables as part of loop initialisation as traditional for loop does. I don't want to create attrMap each time I loop. That is why attrMap has been created outside the loop.

Reply:
Was just trying to optimize, will change.